src/data/set/pointwise/iterate.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import data.set.pointwise.basic
+import algebra.hom.iterate
+import dynamics.fixed_points.basic
+
+/-!
+# Results about pointwise operations on sets with iteration.
+-/
+
+open_locale pointwise
+open set function
+
+/-- Let `n : ℤ` and `s` a subset of a commutative group `G` that is invariant under preimage for
+the map `x ↦ x^n`. Then `s` is invariant under the pointwise action of the subgroup of elements
+`g : G` such that `g^(n^j) = 1` for some `j : ℕ`. (This subgroup is called the Prüfer subgroup when
+ `G` is the circle and `n` is prime.) -/
+@[to_additive "Let `n : ℤ` and `s` a subset of an additive commutative group `G` that is invariant
+under preimage for the map `x ↦ n • x`. Then `s` is invariant under the pointwise action of the
+subgroup of elements `g : G` such that `(n^j) • g = 0` for some `j : ℕ`. (This subgroup is called
+the Prüfer subgroup when `G` is the circle `n` and is prime.)"]
```

Comment:
This is the additive subgroup, not a regular subgroup. Up to you whether to keep the parenthesized comment or drop it.
```suggestion
additive subgroup of elements `g : G` such that `(n^j) • g = 0` for some `j : ℕ`. (This additive subgroup is called
the Prüfer subgroup when `G` is the `add_circle` and `n` is prime.)"]
```

Reply:
OK done!